node/test/Test/Pos/CborSpec.hs

```diff
@@ -223,11 +232,12 @@ roundtripProperty (input :: a) = ((deserialize . serialize $ input) :: a) === in
 -- without breaking anything. This should work with every time which adopted
 -- the schema of having at least one constructor of the form:
 -- .... | Unknown Word8 ByteString
-extensionProperty :: forall a. (Arbitrary a, Eq a, Show a, Bi a) => Property
-extensionProperty = forAll @a (arbitrary :: Gen a) $ \input ->
+extensionPropertyOn :: forall a b. (Eq a, Show a, Bi a, Arbitrary b, Eq b, Show b, Bi b)
+                    => Property
+extensionPropertyOn = forAll @b (arbitrary :: Gen b) $ \input ->
     let serialized      = serialize input -- We now have a BS blob
-        (u :: U)        = deserialize serialized
-        (encoded :: a)  = deserialize (serialize u)
+        (u :: a)        = deserialize serialized
+        (encoded :: b)  = deserialize (serialize u)
```

Comment:
Actually, after thinking a bit more about this, I have decided to postpone the documentation bit as part of this PR:

https://github.com/input-output-hk/cardano-sl/pull/1404

The reason is that this PR is a bit "in-the-flux", because due to the fact we are not using the "tag 24" patter consistently I had to modify the extensionProperty to accept both `U` and `U24`.  In that PR I have unified everything, which means I can write a consistent piece of documentation without confusing the reader even more. Bear with me, will get this fixed as soon as I can 😉